wordpress-mobile/WordPress-Android
# Issue/prompt card visibility logic

This PR adds a logic for regular and compact card visibility.

Branch in this PR is based on #16697 which needs to be merged first.

To test:
- Compact card should be visible on non self-hosted sites.

- Using site that is a blog (has posts on main page) confirm that the prompt is visible.
- Set the Blogging Prompt switch to ON in reminder settings. Confirm that the prompt card is visible.
- Set the Blogging Prompt switch to OFF in reminder settings. Confirm that the prompt card is visible.

- Using site that is not blog (has static main page (portfolio, etc)) confirm that the prompt is not visible..
- Set the Blogging Prompt switch to ON in reminder settings. Confirm that the prompt card is visible.
- Set the Blogging Prompt switch to OFF in reminder settings. Confirm that the prompt card is NOT visible.

## Regression Notes
1. Potential unintended areas of impact
Card visibility logic is not working correctly.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Logic testing.


PR submission checklist:
